### PR TITLE
Sets GO_ENV before tests to be "test"

### PR DESCRIPTION
**Pull request by paganotoni**

### CODE REVIEW

**Review by markbates — CHANGES_REQUESTED** [review]
you should make this another plugin in the `cli/cmds/test` package.

File: cli/internal/plugins/env/go_env.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+}
+
+//BeforeTest should be invoked before tests run to set the GO_ENV variable
+func (ebt *GoEnv) BeforeTest(ctx context.Context, root string, args []string) error {
+	return os.Setenv("GO_ENV", "test")
+}
+
+//AfterTest should be invoked after tests run to reset GO_ENV variable
+func (ebt *GoEnv) AfterTest(ctx context.Context, root string, args []string, err error) error {
+	return os.Setenv("GO_ENV", "")
```
Comment (markbates) [inline cli/internal/plugins/env/go_env.go]: this should set it back to the original value, not clear it.

File: cli/cmds/test/go_env.go [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+}
+
+//BeforeTest should be invoked before tests run to set the GO_ENV variable
+func (ebt *GoEnv) BeforeTest(ctx context.Context, root string, args []string) error {
+	return os.Setenv("GO_ENV", "test")
+}
+
+//AfterTest should be invoked after tests run to reset GO_ENV variable
+func (ebt *GoEnv) AfterTest(ctx context.Context, root string, args []string, err error) error {
+	return os.Setenv("GO_ENV", "")
```
Comment (markbates) [inline cli/cmds/test/go_env.go]: this should revert the `GO_ENV` to its original value, not set it to an empty value.

**Review by markbates — CHANGES_REQUESTED** [review]

File: cli/cmds/test/go_env.go (hunk 2 above)
Comment (paganotoni) [reply cli/cmds/test/go_env.go]: All set, thanks for the feedback.

**Review by markbates — APPROVED** [review]

**Pull request merged**